bionic/cache.py

```diff
@@ -809,7 +811,18 @@ class InvalidCacheStateError(Exception):
     """
 
 
-CACHE_SCHEMA_VERSION = 5
+def filename_from_query(query):
```

Comment:
I don't know if this is the best name since it's not really used as a file name but as stem and a dir. How about something like `encoded_descriptor_from_query`?

Reply:
My thinking here is that the point is to return something that would be appropriate for a filename. In some places we use it as a full filename and in others we use it as a stem.

Maybe `valid_filename_from_query`?

Reply:
Where are we using it as a full filename? I see that it is used a dir name in other places but not full filename.

Reply:
Oh, I was counting a directory name as a filename.